app/V1Module/presenters/ExerciseFilesPresenter.php

```diff
@@ -162,21 +174,32 @@ public function actionUploadAdditionalFiles(string $id) {
     }
 
     $files = $this->uploadedFiles->findAllById($this->getRequest()->getPost("files"));
-    $additionalFiles = [];
+    $newAdditionalFiles = [];
+    $currentAdditionalFiles = [];
+
+    /** @var AdditionalExerciseFile $file */
+    foreach ($exercise->getAdditionalFiles() as $file) {
+      $currentAdditionalFiles[$file->getName()] = $file;
+    }
 
     /** @var UploadedFile $file */
     foreach ($files as $file) {
       if (get_class($file) !== UploadedFile::class) {
         throw new ForbiddenRequestException("File {$file->getId()} was already used somewhere else");
       }
 
-      $additionalFiles[] = $exerciseFile = AdditionalExerciseFile::fromUploadedFile($file, $exercise);
+      if (array_key_exists($file->getName(), $currentAdditionalFiles)) {
+        $currentFile = $currentAdditionalFiles[$file->getName()];
+        $exercise->getAdditionalFiles()->removeElement($currentFile);
+      }
+
+      $newAdditionalFiles[] = $exerciseFile = AdditionalExerciseFile::fromUploadedFile($file, $exercise);
```

Comment:
Why are the files accumulated here, if $newAdditionalFiles is not used afterwards?